vector/src/main/java/im/vector/app/features/analytics/metrics/VectorPlugins.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.analytics.metrics
+
+import im.vector.app.features.analytics.metrics.sentry.SentryDownloadDeviceKeysMetrics
+import org.matrix.android.sdk.api.metrics.MetricPlugin
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Class that contains the all plugins which can be used for tracking.
+ */
+@Singleton
```

Comment:
is it really singleton or more a per session scope?
What happens if you signout signin with a new account? Or maybe it's handled by the plugin?

Reply:
It's singleton. If you signout, the Sentry plugin is stopped and once you sign-in and user consent is given, Sentry plugin is initialized. Otherwise it won't send any events.